src/pseudopeople/noise_functions.py

```diff
@@ -46,6 +43,81 @@ def omit_rows(
     return noised_data
 
 
+def _get_census_omission_noise_levels(
+    population: pd.DataFrame,
+    base_probability: float = data_values.DO_NOT_RESPOND_BASE_PROBABILITY,
+) -> pd.Series:
+    """
+    Helper function for do_not_respond noising based on demography of age, race/ethnicity, and sex.
+
+    :param population: a dataset containing records of simulants
+    :param base_probability: base probability for do_not_respond
+    :return: a pd.Series of probabilities
+    """
+    probabilities = pd.Series(base_probability, index=population.index)
+    probabilities += (
+        population["race_ethnicity"]
+        .astype(str)
+        .map(data_values.DO_NOT_RESPOND_ADDITIVE_PROBABILITY_BY_RACE)
+    )
+    for sex in ["Female", "Male"]:
+        sex_mask = population["sex"] == sex
+        age_bins = pd.cut(
+            x=population[sex_mask]["age"],
+            bins=data_values.DO_NOT_RESPOND_ADDITIVE_PROBABILITY_BY_SEX_AGE[sex].index,
+        )
+        probabilities[sex_mask] += age_bins.map(
+            data_values.DO_NOT_RESPOND_ADDITIVE_PROBABILITY_BY_SEX_AGE[sex]
+        ).astype(float)
+    probabilities[probabilities < 0.0] = 0.0
+    probabilities[probabilities > 1.0] = 1.0
+    return probabilities
+
+
+def apply_do_not_respond(
+    dataset_name: str,
+    dataset_data: pd.DataFrame,
+    configuration: ConfigTree,
+    randomness_stream: RandomnessStream,
+) -> pd.DataFrame:
+    """
+    Applies targeted omission based on demographic model for census and surveys.
+
+    :param dataset_name: Dataset object name being noised
+    :param dataset_data:  pd.DataFrame of one of the form types used in Pseudopeople
+    :param configuration: ConfigTree object containing noise level values
+    :param randomness_stream: RandomnessStream object to make random selection for noise
+    :return: pd.DataFrame with rows from the original dataframe removed
+    """
+    required_columns = ("age", "race_ethnicity", "sex")
+    missing_columns = [col for col in required_columns if col not in dataset_data.columns]
+    if len(missing_columns):
+        raise ValueError(
+            f"Dataset {dataset_name} is missing required columns: {missing_columns}"
+        )
+
+    noise_levels = _get_census_omission_noise_levels(dataset_data)
```

Comment:
nit: Add a comment for future-me explaining why you're getting census noise levels in a function that's being applied to non-census datasets as well

Reply:
ACS and CPS aren't really non-census per se, but it's how it's defined in the concept doc. The noising is based off the census number.